### ISSUE TITLE
Port the fix for: ActionResults returned from controller actions rendered as JSON, instead of executed

### ISSUE THREAD

**Issue by Eilon** [aspnet/Mvc#5344]
Port this fix to 1.0.2: https://github.com/aspnet/Mvc/issues/4960

**Comment by Eilon** [comment]
@ryanbrandenburg this patch is **approved**, please port the fix from 1.1.x and ensure it is merged into the correct branch and building as part of the patch train.

**Comment by ryanbrandenburg** [comment]
The PR  to the patch train is out but @pranavkm says we should wait on the build to stabilize before merging.

**Comment by ryanbrandenburg** [comment]
The PR is merged so I'm closing this.
